pkg/capabilities/capabilities.go

```diff
@@ -4,47 +4,73 @@ import (
 	"fmt"
 	"strings"
 
-	"github.com/syndtr/gocapability/capability"
+	"kernel.org/pub/linux/libs/security/libcap/cap"
 )
 
 var (
-	// NewPid2 is not defined on an interface
+	// GetPID is not defined on an interface
 	// therefore it must be patched in for testability
-	NewPid2 = capability.NewPid2
+	GetPID = func(pid int) (Set, error) {
+		return cap.GetPID(pid)
+	}
 )
 
-func CheckRequired(caps capability.Capabilities, reqCaps []capability.Cap) error {
+// Set is an interface for the cap.Set type, created for testability.
+type Set interface {
+	Clear() error
+	GetFlag(vec cap.Flag, val cap.Value) (bool, error)
+	SetFlag(vec cap.Flag, enabled bool, val ...cap.Value) error
+	SetProc() error
+}
+
+func CheckRequired(caps Set, reqCaps []cap.Value) error {
 	for _, c := range reqCaps {
-		if !caps.Get(capability.EFFECTIVE, c) {
-			return fmt.Errorf("insufficient privileges to run: missing CAP_%s", strings.ToUpper(c.String()))
+		exist, err := caps.GetFlag(cap.Effective, c)
+		if err != nil {
+			return err
+		}
```

Comment:
So you are right, it might not be fatal.
I change it so it will return missing capabilities and error (if one is created from the `cap` package).
The reason this might not be fatal is that the drop capabilities function will ensure the capabilities anyway, just won't make any meaningful error message to the user.